Quadriga/src/main/java/edu/asu/spring/quadriga/web/projectblog/AddProjectBlogEntryController.java

```diff
@@ -132,15 +157,60 @@ public ModelAndView addProjectBlogEntryRequest(
         if (result.hasErrors()) {
             model = new ModelAndView("sites/addprojectblogentry");
             model.getModelMap().put("project", project);
+
         } else {
             String username = principal.getName();
             IUser user = userManager.getUser(username);
             projectBlogEntry.setAuthor(user);
 
             projectBlogEntryManager.addNewProjectBlogEntry(projectBlogEntry);
             model = new ModelAndView("redirect:" + "/sites/" + project.getUnixName() + "/projectblog");
+
             model.getModelMap().put("project", project);
         }
         return model;
     }
+
+    /**
+     * This method is used to return a JSON string for visualizing a network
+     * based on the Network id selected from the UI.
+     * 
+     * @param unixName
+     *            Unix Name given for the Project
+     * @param networkId
+     *            Network Id for the network selected from the UI.
+     * @param principal
+     *            principal object which is required to fetch information about
+     *            logged in user.
+     * @param project
+     *            project instance obtained using @InjectProject annotation
+     * @return Returns a JSON string as a response entity based on the network
+     *         selected from the UI.
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 5, userRole = {
+            RoleNames.ROLE_COLLABORATOR_OWNER, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @InjectProjectByName
+    @RequestMapping(value = "sites/{projectUnixName}/visualizenetwork/{networkId}", method = RequestMethod.GET, produces = "text/plain")
+    public ResponseEntity<String> visualizeNetworks(@ProjectIdentifier @PathVariable("projectUnixName") String unixName,
+            @PathVariable("networkId") String networkId, Principal principal,
+            @CheckAccess @InjectProject IProject project, @RequestParam("projectId") String projectId) {
```

Comment:
You will have to write a new aspect to check that. Just write one that works similar to the AccessPolicies but takes an object rather than the id and checks the object directly. You can then use that for all methods in this class. The way it is right now, it's not secure. I could just call the url for one project but with the id of a different project, and this way could have access to a project that I shouldn't have. Does this make sense?